src/commands.cc

```diff
@@ -988,12 +988,25 @@ const CommandDesc add_hook_cmd = {
                          context, flags, prefix, cursor_pos); }),
     [](const ParametersParser& parser, Context& context, const ShellContext&)
     {
+        auto keydesc = parser[2];
         auto descs = enum_desc(Meta::Type<Hook>{});
         auto it = find_if(descs, [&](const EnumDesc<Hook>& desc) { return desc.name == parser[1]; });
         if (it == descs.end())
             throw runtime_error{format("no such hook: '{}'", parser[1])};
 
-        Regex regex{parser[2], RegexCompileFlags::Optimize};
+        const Vector<Hook> key_hooks = {Hook::InsertKey, Hook::NormalKey, Hook::RawKey};
+        if (contains(key_hooks, it->value))
+        {
+            const auto keys = parse_keys(keydesc);// throws a runtime_error on parsing errors
+
+            if (keys.size() != 1)
+                throw runtime_error{format("a single key description is expected, got {}", keys.size())};
+
+            if (keys[0].key >= Key::F1 and keys[0].key <= Key::F12)
+                keydesc = key_to_str(keys[0]);
```

Comment:
Note: I realize this is overkill, as technically all we need is a call to `uppercase()`, however I felt calling `key_to_str()` was more accurate since that's the same function called to normalize keys when they're hit in normal mode.